components/fxa-client/src/oauth.rs

```diff
@@ -93,10 +93,21 @@ impl FirefoxAccount {
     /// * `scopes` - Space-separated list of requested scopes.
     /// * `wants_keys` - Retrieve scoped keys associated with scopes supporting it.
     pub fn begin_oauth_flow(&mut self, scopes: &[&str], wants_keys: bool) -> Result<String> {
-        let mut url = self.state.config.authorization_endpoint()?;
+        let mut url = if self.state.last_seen_profile.is_some() {
+            self.state.config.content_url_path("/oauth/force_auth")?
+        } else {
+            self.state.config.authorization_endpoint()?
+        };
+
         url.query_pairs_mut()
             .append_pair("action", "email")
             .append_pair("response_type", "code");
+
+        if let Some(ref cached_profile) = self.state.last_seen_profile {
+            url.query_pairs_mut()
+                .append_pair("email", &cached_profile.response.email);
+        }
```

Comment:
Should we also add `uid` to the URL here? IIRC Desktop sends both (in case the user has changed their primary email address but this device doesn't know about it yet)

Reply:
adding `uid` part of this caused validation to fail in the content server

Reply:
Hrm, maybe the force-auth-with-uid logic is only implemented for the special webchannel sync flow but not for OAuth. Let's consider a followup to investigate adding it.

Reply:
https://github.com/mozilla/fxa/issues/1795